users/models.py

```diff
@@ -0,0 +1,69 @@
+from django.db import models
+from django.db.models.signals import pre_save
+from django.core.exceptions import ValidationError
+from django.core.validators import validate_email
+from django.utils.translation import gettext_lazy as _
+from django.contrib.auth.models import AbstractUser
+
+
+def validate_email_addr(email):
+    try:
+        validate_email(email)
+    except ValidationError:
+        raise ValidationError('Email is not valid')
+
+
+def validate_phone_number(phone_number):
+    if phone_number is not None:
+        if phone_number[0] == '+':
+            if phone_number[:4] != '+972':
+                raise ValidationError(
+                    _(f'{phone_number} is not supported'),
+                    params={'phone_number': phone_number},
+                )
+            if len(phone_number) != 13:
+                raise ValidationError(
+                    _(f'{phone_number} is not a valid phone number. The number should have 13 digits'),
+                    params={'phone_number': phone_number},
+                )
+            phone_number = phone_number[1:]
```

Comment:
so according to the tests
+972222222222
should work although it's not a Legal PHN
maybe refer to this and add the validation accordingly
https://en.wikipedia.org/wiki/Telephone_numbers_in_Israel#Cellular_and_mobile_devices_area_code_05

Reply:
done. added support to +972 number as well.